client/landing/gutenboarding/components/domain-picker/index.tsx

```diff
@@ -62,56 +60,21 @@ export interface Props {
 }
 
 const DomainPicker: FunctionComponent< Props > = ( {
-	defaultQuery,
 	onDomainSelect,
 	onDomainPurchase,
 	onClose,
-	queryParameters,
 	currentDomain,
 } ) => {
-	const PAID_DOMAINS_TO_SHOW = 5;
 	const { __: NO__ } = useI18n();
 	const label = NO__( 'Search for a domain' );
 
-	const { siteTitle } = useSelect( select => select( STORE_KEY ).getState() );
-
-	const [ domainSearch, setDomainSearch ] = useState( siteTitle );
-
-	const [ search ] = useDebounce( domainSearch.trim() || defaultQuery || '', selectorDebounce );
-	const searchOptions = {
-		include_wordpressdotcom: true,
-		include_dotblogsubdomain: false,
-		quantity: PAID_DOMAINS_TO_SHOW + 1, // Add our free subdomain
-		...queryParameters,
-	};
-
-	const allSuggestions = useSelect(
-		select => {
-			if ( search ) {
-				return select( DOMAIN_SUGGESTIONS_STORE ).getDomainSuggestions( search, searchOptions );
-			}
-		},
-		[ search, queryParameters ]
-	);
-
-	const freeSuggestions = allSuggestions?.filter( suggestion => suggestion.is_free );
-	const paidSuggestions = allSuggestions
-		?.filter( suggestion => ! suggestion.is_free )
-		.slice( 0, PAID_DOMAINS_TO_SHOW );
+	const { domainSearch } = useSelect( select => select( STORE_KEY ).getState() );
+	const { setDomainSearch } = useDispatch( STORE_KEY );
 
-	// Recommend either an exact match or the highest relevance score
-	const recommendedSuggestion = paidSuggestions?.reduce( ( result, suggestion ) => {
-		if ( result.match_reasons?.includes( 'exact-match' ) ) {
-			return result;
-		}
-		if ( suggestion.match_reasons?.includes( 'exact-match' ) ) {
-			return suggestion;
-		}
-		if ( suggestion.relevance > result.relevance ) {
-			return suggestion;
-		}
-		return result;
-	} );
+	const allSuggestions = useDomainSuggestions();
+	const freeSuggestions = getFreeDomainSuggestions( allSuggestions );
+	const paidSuggestions = getPaidDomainSuggestions( allSuggestions )?.slice( 0, 5 );
```

Comment:
```suggestion
	const paidSuggestions = getPaidDomainSuggestions( allSuggestions )?.slice( 0, PAID_DOMAINS_TO_SHOW );
```

Reply:
Oh, thanks for spotting that one, I'll update it.